explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+subresources to prefetch, the inner resource has to preload the same
+subresources that the aggregator prefetches.
+And the inner response of the main resource signed exchange (article.html.sxg)
+has a preload header and an allowed-alt-sxg header:
+
+```
+Link: <https://cdn.publisher.example/lib.js>;
+        rel="preload";
+        as="script"
+Link: <https://cdn.publisher.example/lib.js>;
+        rel="allowed-alt-sxg";
+        header-integrity="sha256-XXXXXX"
+```
+
+The UA recursively prefetches lib.js.sxg.
+
+If the user navigates to the expected article, both the main resource of the
+article and the script subresource are loaded from the prefetched signed
+exchanges.
+
+# Proposal
+
+## Privacy-preserving prefetching subresources
+
+1. Introduce two new cache mechanism SignedExchangeCache and
+PrefetchedSignedExchangeCache in a Document.
+   - SignedExchangeCache keeps the signed exchanges which are
+prefetched from the Document.
+   - PrefetchedSignedExchangeCache keeps the signed
+exchanges which were passed from the referrer Document which triggered the
+navigation to the current Document.
+1. While processing
+"[prefetch](https://html.spec.whatwg.org/multipage/links.html#link-type-prefetch)"
+link (eg: `<link rel="prefetch"
+href="https://cdn.feed.example/cdn.publisher.example/lib.js.sxg" as="script">`):
+   - If succeeded prefetching the signed exchange (including merkle integrity
+   check of the body), stores the parsedExchange which is the result of
+   [parsing a signed exchange](https://wicg.github.io/webpackage/loading.html#ref-for-parsing-a-signed-exchange)
+   (= inner request URL and inner response) to the SignedExchangeCache.
+1. While
+[navigating across documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents),
+copy the parsedExchanges in the SignedExchangeCache of the source document to
+the PrefetchedSignedExchangeCache of the target document. This is intended to
+provide a way to pass the cached signed exchange across origins even if the UA
+is using origin isolated HTTPCache mechanism.
+1. While processing
+[preload](https://html.spec.whatwg.org/multipage/links.html#link-type-preload)
+link HTTP headers (eg: Link: <https://cdn.publisher.example/lib.js>;
+rel="preload"; as="script"):
+   - Check whether matching "allowed-alt-sxg" link HTTP header
+   (`Link: <https://cdn.publisher.example/lib.js>;rel="allowed-alt-sxg";header-integrity="sha256-.."`)
+   exists or not. (Note that if the allowed-alt-sxg link HTTP header has
+   variants and variant-key attributes, the UA must execute the algorithm
+   written in
+   [HTTP Representation Variants spec](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+   to find the matching header.)
+   - Check whether all preload links which have matching allowed-alt-sxg link
+   header have matching (url and header-integrity) parsedExchange in
+   PrefetchedSignedExchangeCache. If the check passes, set request's stashed
```

Comment:
"request" hasn't been introduced yet.

Reply:
Changed to refer "linked resource fetch setup steps".